src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/RazorSemanticTokensInfoService.cs

```diff
@@ -283,9 +351,16 @@ internal static bool TryGetMinimalCSharpRange(RazorCodeDocument codeDocument, Ra
 
         var data = new int[semanticRanges.Count * TokenSize];
         var semanticRangeCount = 0;
+        var sourceText = razorCodeDocument.GetSourceText();
+
+        if (sourceText is null)
+        {
+            return Array.Empty<int>();
```

Comment:
If we're going to return early with an empty array, then we should probably it before allocating a giant array on line 352 😁

But realistically, I don't think SourceText can ever be null. Was this maybe due to a quirk in the benchmark setup?

Reply:
Nullability checks suggested adding this; you are correct, earlier is better.

Reply:
Thanks for checking, I missed that this method is in the compiler, and probably null-oblivious.